File: app/src/main/java/info/nightscout/androidaps/utils/Profiler.java
package info.nightscout.androidaps.utils;

import org.slf4j.Logger;

import info.nightscout.androidaps.logging.AAPSLogger;
import info.nightscout.androidaps.logging.LTag;

/**
 * Created by mike on 29.01.2017.
 */

public class Profiler {
    public Profiler() {
    }

    @Deprecated
    static public void log(Logger log, String function, long start) {
        long msec = System.currentTimeMillis() - start;
        log.debug(">>> " + function + " <<< executed in " + msec + " miliseconds");
    }

    static public void log(AAPSLogger log, LTag lTag, String function, long start) {
        long msec = System.currentTimeMillis() - start;
        log.debug(lTag, ">>> " + function + " <<< executed in " + msec + " miliseconds");
    }
}

File: app/src/main/java/info/nightscout/androidaps/utils/Profiler.kt
package info.nightscout.androidaps.utils

import info.nightscout.androidaps.annotations.OpenForTesting
import info.nightscout.shared.logging.AAPSLogger
import info.nightscout.shared.logging.LTag
import javax.inject.Inject
import javax.inject.Singleton

@OpenForTesting
@Singleton
class Profiler @Inject constructor(val aapsLogger: AAPSLogger) {

    fun log(lTag: LTag, function: String, start: Long) {
        val milliseconds = System.currentTimeMillis() - start
        aapsLogger.debug(lTag, ">>> $function <<< executed in $milliseconds milliseconds")
    }
}